### ISSUE TITLE
the param of T is computed by minimizing the ce loss.

### ISSUE THREAD

**Issue by KeyKy** [gpleiss/temperature_scaling#3]
1.So, Why not fix the T and train the convolution params?
2.Why minimize ce loss not the ECE loss? Is there a proof of minimizing ce loss equal to minimizing ECE loss?
Thanks!

**Comment by zihaolucky** [comment]
The ECE is discrete. To minimize the CE loss, the false classified sample might reduce their degree of error which reflect on its final score(`confidence`)

**Comment by gpleiss** [comment]
Also see the FAQ section here: http://geoffpleiss.com/nn_calibration

**Comment by ShellingFord221** [comment]
Hi, in the paper, we should optimize temperature by minimizing NLL loss. However, ECE loss is actually what we want: to make confidence = accuracy on validation set. So can I ask why calibration is associated with NLL loss rather than ECE loss? Thanks!

**Comment by gpleiss** [comment]
See the answer above. Also, ECE is a brittle metric (since it relies on binning), whereas NLL is more continuous. Since temperature scaling doesn't change accuracy, any improvement in NLL is directly an improvement in calibration and therefore ECE.
